dataplane/dataplane.cpp

```diff
@@ -443,6 +450,7 @@ eResult cDataPlane::initPorts()
 		tPortId portId;
 		if (strncmp(name.data(), SOCK_DEV_PREFIX, strlen(SOCK_DEV_PREFIX)) == 0)
 		{
+			YANET_LOG_INFO("Opening sockdev with path %s\n", name.data() + strlen(SOCK_DEV_PREFIX));
 			portId = sock_dev_create(name.data() + strlen(SOCK_DEV_PREFIX), interfaceName.c_str(), 0);
 		}
 		else if (rte_eth_dev_get_port_by_name(name.data(), &portId))
```

Comment:
oh no, you add integer to a `char*` here which shifts it. I understand why you do this, but we can do it so much better:

First of all, let's use `string::rfind` with an offset of 0 to check if the name starts with the prefix.
Then, instead of shifting the pointer, we can just create a new string.
Or even better, just create a non-owning `std::string_view`:
```cpp
        tPortId portId = 0;
        std::string_view name_view(name);

        if (name_view.starts_with(SOCK_DEV_PREFIX))
        {
            std::string_view sockdev_path = name_view.substr(strlen(SOCK_DEV_PREFIX));
            YANET_LOG_INFO("Opening sockdev with path %.*s\n", static_cast<int>(sockdev_path.size()), sockdev_path.data());
            portId = sock_dev_create(sockdev_path.data(), interfaceName.c_str(), 0);
        }
        else if (rte_eth_dev_get_port_by_name(name.c_str(), &portId) != 0)
        {
        //...
```